Gems/ScriptCanvas/Code/Editor/Assets/ScriptCanvasFileHandling.cpp

```diff
@@ -249,13 +277,41 @@ namespace ScriptCanvasEditor
         if (auto entity = scriptCanvasData->GetScriptCanvasEntity())
         {
             AZ_Assert(entity->GetState() == AZ::Entity::State::Constructed, "Entity loaded in bad state");
-            AZ::u64 entityId =
-                aznumeric_caster(ScriptCanvas::MathNodeUtilities::GetRandomIntegral<AZ::s64>(1, std::numeric_limits<AZ::s64>::max()));
-            entity->SetId(AZ::EntityId(entityId));
+
+            auto entityIdTypeId = azrtti_typeid<AZ::EntityId>();
+            ScriptCanvasFileHandlingCpp::EntityIdMapper entityMapper;
+            entity->SetId(entityMapper.GetNewId(entity->GetId()));
+            
+            // Create new EntityIds for all EntityIds found in the SC Entity/Component objects
+            // and map all old Ids to the new ones. This way, no Entity activation/deactivation, or
+            // bus communication via EntityId will be handled by multiple or incorrect objects on
+            // possible multiple instantiations of graphs.
+            //
+            // EntityIds contained in variable (those set to self or the graph unique id, will be ignored)
+            auto beginElementCB = [&entityMapper, &entityIdTypeId]
```

Comment:
what is CB, maybe full name for readbility?

just want to double check this workflow only happens when deserialize from file, right? so each graph will have a different sc entity/component id? what will happen when we serialize the graph into file? will use new ids or still old ids?

Side question: we can open multiple instantiations of same graph?

Reply:
CB is "call back", used throughout the engine. 

This is required because we can and do open multiple instantiations of the same graph. This does only happen we we deserialize.  No changes are required for when we serialize.